### ISSUE TITLE
"DELETE FROM ... USING .." not available in sqlite (quickstart)

### ISSUE THREAD

**Issue by Tigge** [jensl/critic#105]
There are SQL statements in critic on the form `DELETE FROM [...] USING [...]` which is not supported in SQLite.

There are a few of these around the code base:
```
src/background/changeset.py:115
src/background/highlight.py:160
src/index.py:484
src/dbutils/user.py:94
src/operation/manipulateassignments.py:79
src/operation/manipulatecomment.py:268
src/operation/manipulatecomment.py:276
```

These could be done using nested selects instead. Like this:

```diff
diff --git a/src/operation/manipulatecomment.py b/src/operation/manipulatecomment.py
index 7cdc363..1c0ac4b 100644
--- a/src/operation/manipulatecomment.py
+++ b/src/operation/manipulatecomment.py
@@ -265,10 +265,11 @@ class MarkChainsAsRead(Operation):
 
         if chain_ids:
             cursor.execute("""DELETE FROM commentstoread
-                                    USING comments
                                     WHERE commentstoread.uid=%s
-                                      AND commentstoread.comment=comments.id
-                                      AND comments.chain=ANY (%s)""",
+                                      AND commentstoread.comment IN
+                                  (SELECT id
+                                     FROM comments
+                                    WHERE comments.chain=ANY (%s))""",
                            (user.id, chain_ids))
 
         if review_ids:

```

I could do a run and adjust these, or is there another prefererad solution to this problem?

**Comment by jensl** [comment]
Yeah, they should be fixed, and that's the way to fix them, AFAIK. I fixed a bunch of them in aba5c98e, but was obviously quite lazy when searching for queries to fix.

If you were to upload a review with the rest fixed, I'd be ever so happy to review it. :-)

**Comment by Tigge** [comment]
Pushed for review: https://critic-review.org/r/358
